clients/src/main/java/org/apache/kafka/common/requests/AbstractRequest.java

```diff
@@ -76,54 +88,78 @@ public short version() {
     /**
      * Factory method for getting a request object based on ApiKey ID and a buffer
      */
-    public static AbstractRequest getRequest(int requestId, short versionId, ByteBuffer buffer) {
+    public static RequestAndSize getRequest(int requestId, short version, ByteBuffer buffer) {
```

Comment:
I had that it that way first, but it's a bit weird because we don't always need the size and we can't compute it unless the `Struct` constructor is used. This way is reasonably clean and the cost seems low. What do you think?